### PR TITLE
Deprecate 'in ref' to prepare for -preview=in

### PR DESCRIPTION
**Pull request by Geod24**
This should nudge people into using '-preview=in' and clear up the remaining projects which are incompatible with it.

### CODE REVIEW

**Review by thewilsonator — COMMENTED** [review]
Apart from the failing test and lack of changelog entry this looks good.

File: compiler/test/fail_compilation/deprecatedinref.d [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+REQUIRED_ARGS: -de
+TEST_OUTPUT:
+---
+fail_compilation/fail9735.d(10): Deprecation: casting from void delegate() to void* is deprecated
```
[the remark this answers is not in the document]
Comment (Geod24) [reply compiler/test/fail_compilation/deprecatedinref.d]: Oops, fixed

**Comment by Geod24** [comment]
Fixed tests. needs https://github.com/dlang/phobos/pull/8703

**Comment by Geod24** [comment]
Added changelog. @thewilsonator I was looking into adding a spec change, but actually, this is already in the spec:
image

As `in` and `ref` are defined as being mutually exclusive.

**Review by thewilsonator — APPROVED** [review]

**Comment by Geod24** [comment]
CC @atilaneves @WalterBright

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**